src/App.js

```diff
@@ -1,13 +1,13 @@
 import React from "react";
 import NameForm from "./components/NameForm";
+import Chart from "./components/Chart";
+import ComplexView from "./components/ComplexView";
 
 const App = () => (
     <div className="App">
-        <NameForm callback={(number) => {
-            alert(number);
-        }}/>
-        <Chart/>
+        <ComplexView  >{(callback)=>[<NameForm callback={callback}/>, <Chart values={[[new Date(), 20.20], [new Date(), 21.20]]}/>]}</ComplexView>
```

Comment:
It is cool here.
Just I wonder if it is possible to move with it forward.

The next step would be to create a StatedNameForm, however StatedNameForm should not be taking any props, just using Services to change pages and to get state. Therefore there won't be no callback for ComplexView to use.

I think it is time to move to router unfortunately.